bin/install

```diff
@@ -1,158 +1,449 @@
+    or abort "No installation detected at $OMF_PATH"
+
+  say "Warning: This will uninstall Oh My Fish and all plugins and themes installed in $OMF_PATH. Your configuration will not be modified. You will need to remove any Oh My Fish startup code from $FISH_CONFIG/config.fish."
+
+  confirm_yes "Are you sure you want to continue?"
+  say "Uninstalling from $OMF_PATH..."
+
+  # Remove the core framework
+  command rm -rf "$OMF_PATH/.git"
+    and command rm -r "$OMF_PATH"
+    or abort "Uninstall failed"
 
-  if not available git
-    report error "Aborting: Installation requires Git."
+  # Remove the bootstrap if it is managed by us
+  set -l vendor_config_file "$FISH_CONFIG/conf.d/omf.fish"
+  test -e "$vendor_config_file"
+    and command rm "$vendor_config_file"
+    or abort "Failed to remove bootstrap file"
+
+  say "Uninstall complete"
+end
+
+
+# Makes a backup of a given file.
+function backup_file -a file_path
+  test -e "$file_path"; or return 1
+
+  set -l path (command dirname $file_path)
+  set -l file (command basename $file_path)
+  set -l name (echo $file | command cut -d. -f1)
+
+  set -l timestamp (command date +%s)
+  set -l backup_file "$path/$name.$timestamp.copy"
+
+  say "Existent $file found at $path"
+  say "↳ Moving to $backup_file"
+
+  if not command mv "$file_path" $backup_file 2>/dev/null
+    abort "Could not backup $file_path"
   end
 
-  if not fish_version_compatible
-    set -l minimum_version_string (echo $OMF_FISH_MIN_VER | sed 's/ /./g')
-    report error "Aborting: Detected fish $version, but Oh My Fish requires fish $minimum_version_string or greater."
+  return 0
+end
+
+
+# Gets the version of Fish installed.
+function get_fish_version
+  set -q FISH_VERSION
+    and echo $FISH_VERSION
+    or echo $version
+end
+
+
+# Gets the version of Git installed.
+function get_git_version
+  type -f -q git
+    and command git --version | command cut -d' ' -f3
+end
+
+
+# Checks if a path looks like an OMF install.
+function is_install_dir -a path
+  test -n "$path"
+    and test -d "$path"
+    and test -d "$path/pkg/omf"
+end
+
+
+# Checks if OMF is set up properly and working.
+function is_omf_loaded
+  command fish -c "omf --version" > /dev/null ^&1
+end
+
+
+# Tests if the right-hand side version is equal to or greater than the left-hand side version.
+function is_version_compatible -a lhs rhs
+  test -n "$lhs" -a -n "$rhs" -a "$lhs" = (printf "$lhs\n$rhs\n" | command sort -t '.' -k 1,1 -k 2,2 -k 3,3 -k 4,4 -g)[1]
+end
+
+
+# Assert that all tools we need are available.
+function assert_cmds
+  set -l cmds basename cp cut date dirname fish fold git mkdir mv sed sort tar
+
+  for cmd in $cmds
+    type -f -q $cmd
+      or abort "Command '$cmd' not found"
   end
+end
+
+
+# Assert that a minimum required version of Fish is installed.
+function assert_fish_version_compatible -a required_version
+  set -l installed_version (get_fish_version)
+    and is_version_compatible $required_version $installed_version
```

Comment:
`get_fish_version` should `return 1` for this `and` clause make sense. This change should catch the edge case of neither `FISH_VERSION` and `version` variables exist.